File: src/main/java/top/zbeboy/isy/web/util/AjaxUtils.java
package top.zbeboy.isy.web.util;


import lombok.RequiredArgsConstructor;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;

import java.util.List;
import java.util.Map;

/**
 * Created by lenovo on 2016-01-09.
 * ajax消息以及数据封装
 */
@Slf4j
@RequiredArgsConstructor(staticName = "of")
@ToString
public class AjaxUtils<T> {

    private boolean state;//消息状态
    private String msg;//消息
    private Map<String, Object> mapResult;//map数据
    private List<T> listResult;//list数据
    private Object objectResult;//单个对象数据
    private PaginationUtils paginationUtils;//分页数据

    public AjaxUtils<T> success() {
        this.state = true;
        return this;
    }

    public AjaxUtils<T> fail() {
        this.state = false;
        return this;
    }

    public AjaxUtils<T> msg(String msg) {
        this.msg = msg;
        return this;
    }

    public AjaxUtils<T> obj(Object obj) {
        this.objectResult = obj;
        return this;
    }

    public AjaxUtils<T> mapData(Map<String, Object> map) {
        this.mapResult = map;
        return this;
    }

    public AjaxUtils<T> listData(List<T> list) {
        this.listResult = list;
        return this;
    }

    public AjaxUtils<T> paginationUtils(PaginationUtils paginationUtils) {
        this.paginationUtils = paginationUtils;
        return this;
    }

    public boolean isState() {
        return state;
    }

    public void setState(boolean state) {
        this.state = state;
    }

    public PaginationUtils getPaginationUtils() {
        return paginationUtils;
    }

    public void setPaginationUtils(PaginationUtils paginationUtils) {
        this.paginationUtils = paginationUtils;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public Object getObjectResult() {
        return objectResult;
    }

    public void setObjectResult(Object obj) {
        this.objectResult = obj;
    }

    public Map<String, Object> getMapResult() {
        return mapResult;
    }

    public void setMapResult(Map<String, Object> mapResult) {
        this.mapResult = mapResult;
    }

    public List<T> getListResult() {
        return listResult;
    }

    public void setListResult(List<T> listResult) {
        this.listResult = listResult;
    }
}

File: src/main/java/top/zbeboy/isy/web/util/AjaxUtils.kt
package top.zbeboy.isy.web.util

/**
 * Created by zbeboy 2017-11-03 .
 * ajax消息以及数据封装
 **/
open class AjaxUtils<T> {

    var state: Boolean = false//消息状态
    var msg: String? = null//消息
    var mapResult: Map<String, Any>? = null//map数据
    var listResult: List<T>? = null//list数据
    var objectResult: Any? = null//单个对象数据
    var paginationUtils: PaginationUtils? = null//分页数据

    companion object {
        @JvmStatic
        fun <T> of(): AjaxUtils<T> {
            return AjaxUtils()
        }
    }

    fun success(): AjaxUtils<T> {
        this.state = true
        return this
    }

    fun fail(): AjaxUtils<T> {
        this.state = false
        return this
    }

    fun msg(msg: String): AjaxUtils<T> {
        this.msg = msg
        return this
    }

    fun obj(obj: Any): AjaxUtils<T> {
        this.objectResult = obj
        return this
    }

    fun mapData(map: Map<String, Any>): AjaxUtils<T> {
        this.mapResult = map
        return this
    }

    fun listData(list: List<T>): AjaxUtils<T> {
        this.listResult = list
        return this
    }

    fun paginationUtils(paginationUtils: PaginationUtils): AjaxUtils<T> {
        this.paginationUtils = paginationUtils
        return this
    }
}